lucet-concurrency-tests/src/killswitch.rs

```diff
@@ -478,6 +478,152 @@ macro_rules! killswitch_tests {
             })
         }
 
+        // If we terminate in the signal handler, but before termination has been disabled, a
+        // signal will be sent to the guest. Lucet must correctly handle this case, lest the sigalrm be
+        // delivered to disastrous effect to the host.
+        //
+        // This corresponds to a race during the documentation's State B -> State E "guest faults
+        // or is terminated" transition.
+        #[test]
+        fn terminate_during_guest_fault() {
+            test_c_with_instrumented_guest_entry("timeout", "fault.c", |mut inst| {
+                let kill_switch = inst.kill_switch();
+
+                // *before* termination is critical, since afterward the `KillSwitch` we test with will
+                // just take no action.
+                let unfortunate_time_to_terminate = inst
+                    .lock_testpoints
+                    .signal_handler_before_disabling_termination
+                    .wait_at();
+                // wait for the guest to reach a point we reaaallly don't want to signal at - somewhere in
+                // the signal handler.
+                let exiting_signal_handler = inst
+                    .lock_testpoints
+                    .signal_handler_before_returning
+                    .wait_at();
+                // and we need to know when we're ready to signal to ensure it races with.
```

Comment:
I think think the sentence in the comment might be incomplete? Forgive me if I'm just misreading this.